streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -389,6 +391,18 @@ public void close() {
         metrics.removeAllSensors();
     }
 
+    void closeProducer() {
```

Comment:
Would it not be better for this to be called `close`? It's odd to have a `closeProducer` method that calls `recordCollector.close`. It seems like a layering violation.